src/blockdev.rs

```diff
@@ -48,7 +48,11 @@ pub struct Disk {
 impl Disk {
     pub fn new(path: &str) -> Self {
         Disk {
-            path: path.to_string(),
+            path: if let Ok(target) = canonicalize(path) {
+                target.to_string_lossy().to_string()
+            } else {
+                path.to_string()
+            },
```

Comment:
I don't think we should gracefully handle `canonicalize` failing here. I.e. let's just change `new` to return a `Result`?

Reply:
Sure, will get to that tomorrow :+1:

Reply:
Commit updated as requested